### ISSUE TITLE
Internal Bars when no data

### ISSUE THREAD

**Issue by rsmb7z** [nautechsystems/nautilus_trader#868]
# Bug Report

### Expected Behavior
Bar should not triger when there is not data available during subscription i.e. Closed market. Currently bar returns with pricing of -1.

### Actual Behavior
When subscribing QuoteTicks and Internal Aggregation bars.

### Steps to Reproduce the Problem

1. Subscribe to Live subscribe_quote_ticks or subscribe_trade_ticks
2. Subscribe any bar with Internal Aggregation

### Specifications

- OS platform: Win10
- Python version: 3.10
- `nautilus_trader` version: 1.158.0 and 1.159.0

**Comment by cjdsellers** [comment]
Are you able to provide logs of the `-1` price?

I agree there is currently no mechanism to only 'print' bars during specified hours though. I believe this would be an enhancement request for the bar aggregator though, rather than a bug.

**Comment by rsmb7z** [comment]
Hi @cjdsellers 
As you can see the bars are actually being receiving with price of -1 and indicator is processing it as well. What I meant is aggregator shouldn't send the Bars if there is no data. So this will likely be bug causing unexpected calculations to the indicator and ultimately strategy conditions which is depending to indicators. This can be tested during market closed. Here are the logs:

```
2022-11-19T09:17:14.997492600Z [INF] TESTER-001.SID-OID: Bar(EUR.USD@IDEALPRO.IB-5-SECOND-MID-INTERNAL,-1.000000,-1.000000,-1.000000,-1.000000,0.0,1668849435000000200)
2022-11-19T09:17:14.997498600Z [INF] TESTER-001.SID-OID: Waiting for indicators to warm up [16]...
2022-11-19T09:17:18.404002000Z [DBG] TESTER-001.ExecEngine: Checking in-flight orders status...
2022-11-19T09:17:18.404008200Z [DBG] TESTER-001.ExecEngine: Found 0 orders in-flight.
2022-11-19T09:17:19.997286400Z [INF] TESTER-001.SID-OID: Bar(EUR.USD@IDEALPRO.IB-5-SECOND-MID-INTERNAL,-1.000000,-1.000000,-1.000000,-1.000000,0.0,1668849440000000200)
2022-11-19T09:17:19.997292000Z [INF] TESTER-001.SID-OID: Waiting for indicators to warm up [17]...
2022-11-19T09:17:23.404873300Z [DBG] TESTER-001.ExecEngine: Checking in-flight orders status...
2022-11-19T09:17:23.404879000Z [DBG] TESTER-001.ExecEngine: Found 0 orders in-flight.
2022-11-19T09:17:24.998262400Z [INF] TESTER-001.SID-OID: Bar(EUR.USD@IDEALPRO.IB-5-SECOND-MID-INTERNAL,-1.000000,-1.000000,-1.000000,-1.000000,0.0,1668849445000000200)
2022-11-19T09:17:24.998268300Z [INF] TESTER-001.SID-OID: Waiting for indicators to warm up [18]...
2022-11-19T09:17:28.404865300Z [DBG] TESTER-001.ExecEngine: Checking in-flight orders status...
2022-11-19T09:17:28.404871400Z [DBG] TESTER-001.ExecEngine: Found 0 orders in-flight.
2022-11-19T09:17:29.996973300Z [INF] TESTER-001.SID-OID: Bar(EUR.USD@IDEALPRO.IB-5-SECOND-MID-INTERNAL,-1.000000,-1.000000,-1.000000,-1.000000,0.0,1668849450000000200)
2022-11-19T09:17:29.996980300Z [INF] TESTER-001.SID-OID: Waiting for indicators to warm up [19]...
2022-11-19T09:17:33.404246100Z [DBG] TESTER-001.ExecEngine: Checking in-flight orders status...
2022-11-19T09:17:33.404251800Z [DBG] TESTER-001.ExecEngine: Found 0 orders in-flight.
2022-11-19T09:17:34.997154800Z [INF] TESTER-001.SID-OID: Bar(EUR.USD@IDEALPRO.IB-5-SECOND-MID-INTERNAL,-1.000000,-1.000000,-1.000000,-1.000000,0.0,1668849455000000200)
2022-11-19T09:17:34.997160800Z [INF] TESTER-001.SID-OID: Waiting for indicators to warm up [20]...
2022-11-19T09:17:38.404519900Z [DBG] TESTER-001.ExecEngine: Checking in-flight orders status...
2022-11-19T09:17:38.404525400Z [DBG] TESTER-001.ExecEngine: Found 0 orders in-flight.
2022-11-19T09:17:39.997394700Z [INF] TESTER-001.SID-OID: Bar(EUR.USD@IDEALPRO.IB-5-SECOND-MID-INTERNAL,-1.000000,-1.000000,-1.000000,-1.000000,0.0,1668849460000000200)
2022-11-19T09:17:43.403734900Z [DBG] TESTER-001.ExecEngine: Checking in-flight orders status...
2022-11-19T09:17:43.403740700Z [DBG] TESTER-001.ExecEngine: Found 0 orders in-flight.
2022-11-19T09:17:44.997228100Z [INF] TESTER-001.SID-OID: Bar(EUR.USD@IDEALPRO.IB-5-SECOND-MID-INTERNAL,-1.000000,-1.000000,-1.000000,-1.000000,0.0,1668849465000000200)
2022-11-19T09:17:48.404890500Z [DBG] TESTER-001.ExecEngine: Checking in-flight orders status...
2022-11-19T09:17:48.404896300Z [DBG] TESTER-001.ExecEngine: Found 0 orders in-flight.
```

**Comment by cjdsellers** [comment]
I believe we have now fixed this from #928.
